### PR TITLE
Remove icmp

### PR DESCRIPTION
**Pull request by klarose**
This PR attempts to deal with issue #8 by replacing references to ICMP with a generic signaling protocols, then specifying requirements for it.

### CODE REVIEW

File: draft-ietf-capport-architecture.xml [hunk 1]
```diff
@@ -401,18 +398,54 @@
...
-         that packets could not be delivered (network unreachable, port
-         unreachable, packet too large, etc.).
+        An Enforcement Device may also want to inform the User Equipment of a pending expiry
+        of its access to the external network, so providing the Enforcement Device the ability
+        to preemptively signal may be desireable.
+       </t>
+       <t>
+        A specific Captive Portal Signalling Protocol is out of scope for this document.
+        However, in order to ensure that future protocols fit into the architecture,
+        requirements for a Captive Portal Signalling Protocl follow:
```
Comment (martinthomson) [inline draft-ietf-capport-architecture.xml]: typo: Protocl

File: draft-ietf-capport-architecture.xml [hunk 2]
```diff
@@ -401,18 +398,54 @@
...
+          <list style="numbers">  
+            <t>The notification SHOULD NOT be easy to spoof.
+               If an attacker can send spoofed notifications to the User Equipment,
+               they can cause the User Equipment to unnecessarily access the API. Rather than
+               relying solely on rate limits to prevent problems, a good protocol will strive
+               to limit the feasibility of such attacks.
+            </t>
+            <t>It SHOULD be possible to send the notification before the captive portal closes.
+               This will help ensure seemless connectivity for the user, as the User Equipment
```
Comment (martinthomson) [inline draft-ietf-capport-architecture.xml]: seamless

File: draft-ietf-capport-architecture.xml [hunk 3]
```diff
@@ -401,18 +398,54 @@
...
+               will not need to wait for a network failure to refresh its login. On receipt of
+               preemptive notication, the User Equipment can prompt the user to refresh.
+             </t>
+             <t>The signal SHOULD be a simple binary protocol indicating that the user equipment
+                is either captive or soon to be captive.
```
Comment (martinthomson) [inline draft-ietf-capport-architecture.xml]: This isn't really a requirement, it's desiderata, sure, but the point I think that you wanted to capture is that the protocol not include any information other than a prompt to contact the API.

File: draft-ietf-capport-architecture.xml [hunk 4]
```diff
@@ -401,18 +398,54 @@
...
+             </t>
+             <t>
+                It is NOT REQUIRED that the signalling protocol be reliable.
```
Comment (martinthomson) [inline draft-ietf-capport-architecture.xml]: "NOT REQUIRED" is not a 2119 keyword, so not sure what capitalizing it does.

US English is "signaling" (something that used to annoy me no end).

**Comment by klarose** [comment]
I missed some stuff in here. My big long list of todos got messed up, so I thought I was done this bit. :/ Basically, some diagrams/etc still need work. I'll take your feedback while finishing it up.

File: draft-ietf-capport-architecture.xml (hunk 4 above)
Comment (klarose) [reply draft-ietf-capport-architecture.xml]: Hmm. I'm trying to point out that that protocol designers need not make the protocol reliable. Perhaps simple omission, plus a note to that effect in the next point is sufficient?

Comment (martinthomson) [reply draft-ietf-capport-architecture.xml]: That's probably better, yeah.

File: draft-ietf-capport-architecture.xml [hunk 5]
```diff
@@ -779,11 +764,13 @@ o . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . o
     </section>
    </section>
 
-   <section anchor="Acknowledgements" title="Acknowledgements">
+   <section anchor="Acknowledgments" title="Acknowledgments">
+     <t>The authors thank Lorenzo Colitti for providing the majority of the content
```
Comment (klarose) [inline draft-ietf-capport-architecture.xml]: Should this be in a Contribution section? I'm not sure of the etiquette here.

Comment (martinthomson) [reply draft-ietf-capport-architecture.xml]: Acknowledgments are good.  Contributors is usually used for substantial editorial contributions, not just ideas and feedback.  But both are left to editor discretion.  Add as you see fit.

**Pull request merged**